src/utils/featureStyleUtils.js

```diff
@@ -203,6 +203,62 @@ export function getTextColor(style) {
     return null
 }
 
+/**
+ * Get text alignment from style *
+ *
+ * @param {Number} textScale Text scaling
+ * @param {FeatureStyleSize} iconSize Icon size parameters
+ * @param {IconStyle} iconStyle KML icon style containing absolute anchor values
+ * @param {IconArgs} iconArgs Parsed url containing name of icon
+ * @returns {Array | null} Returns the feature label offset
+ */
+export function getTextOffset(textScale, iconSize, iconStyle, iconArgs) {
+    const iconScale = iconSize.iconScale
+    const iconName = iconArgs ? iconArgs.name : null
+    const iconAnchor = iconStyle.getAnchor()
+    const iconExtent = iconStyle.getSize()
+    const anchor = [iconAnchor[0] / iconExtent[0], iconAnchor[1] / iconExtent[1]]
+
+    return calculateTextOffset(textScale, iconScale, anchor, iconName)
+}
+
+/**
+ * Calculate text alignment from style parameters *
+ *
+ * @param {Number} textScale Text scaling
+ * @param {Number} iconScale Icon scaling
+ * @param {Array} anchor Relative position of Anchor
+ * @param {String} iconName Name of icon
+ * @returns {Array | null} Returns the feature label offset
+ */
+export function calculateTextOffset(textScale, iconScale, anchor, iconName) {
+    if (!iconScale) {
+        return [0, 0]
+    }
+
+    const iconSize = [32, 32]
```

Comment:
The icon size should not be hardcoded. It is planned to have it from the backend see https://jira.swisstopo.ch/browse/PB-303. 
But why 32 pixel here and not 48 pixel ? Is it because of openlayer using 32 pixel as basis ?

Reply:
You are right it is 48 by 48, I should have used a scaling factor instead of reducing the extent. Since all our icons have quite a big transparency border their apparent size is much smaller, causing the gap between icon and text to grow with increasing icon size, but I have removed that scaling for now.